### DISCUSSION TITLE
Guidance on Integrating Google Login (No signup) only for existing users in Keycloak

### DISCUSSION THREAD

**Discussion by DagdiHiman** [keycloak/keycloak#35264] [Q&A]
Hi All! :wave:
First off, thanks for maintaining such an incredible tool! :rocket:

I’m currently working on a setup where I want to:
   - Pre-create users in Keycloak with just their email and name (no credentials or linked social accounts). I have a set of users and would want only them to have access to the platform and that too only using google sign-in (no signup).
   - Allow these users to log in exclusively via Google (disabling the username/password login entirely).
   - On their first Google login, Keycloak should automatically match the email address and link their Google identity provider (i.e., no manual linking required).
   - Reject login attempts for any users not pre-created/existing in Keycloak.

I’ve already integrated Google Identity Provider, but now I’m stuck on how to:
  - Set up automatic email-based linking of Google identity provider for pre-created/existing users.
  - Ensure username/password is disabled entirely while keeping the flow seamless for end users.

Is this achievable using custom authentication flows, or is there another recommended approach? Any guidance, detailed steps, or tips from those who’ve tackled similar use cases would be greatly appreciated! :pray:

Looking forward to hearing your thoughts—thanks in advance for your time! :bulb:

**Comment by HitKoladiya** [comment]
Have you found the answer because i'm also searching?

**Comment by nguyenvantuan2391996** [comment]
@DagdiHiman @HitKoladiya Login admin dashboad -> Authentication -> Create a flow look like this image. I hope it useful for you
![image](https://github.com/user-attachments/assets/4df73814-282a-4b48-9cf7-41363147de38)
